dask_jobqueue/tests/test_jobqueue_core.py

```diff
@@ -44,3 +44,50 @@ def test_forward_ip():
     with PBSCluster(walltime='00:02:00', processes=4, cores=8, memory='28GB',
                     name='dask-worker') as cluster:
         assert cluster.local_cluster.scheduler.ip == default_ip
+
+
+@pytest.mark.parametrize('Cluster', [PBSCluster, MoabCluster, SLURMCluster,
+                                     SGECluster, LSFCluster])
+@pytest.mark.parametrize(
+    'qsub_return_string',
+    ['{job_id}.admin01',
+     'Request {job_id}.asdf was sumbitted to queue: standard.',
+     'sbatch: Submitted batch job {job_id}',
+     '{job_id};cluster',
+     'Job <{job_id}> is submitted to default queue <normal>.',
+     '{job_id}'])
+def test_job_id_from_qsub(Cluster, qsub_return_string):
+    original_job_id = '654321'
+    qsub_return_string = qsub_return_string.format(job_id=original_job_id)
+    with Cluster(walltime='00:02:00', processes=4, cores=8, memory='28GB',
+                 name='dask-worker') as cluster:
+        assert (original_job_id
+                == cluster._job_id_from_submit_output(qsub_return_string))
+
+
+@pytest.mark.parametrize('Cluster', [PBSCluster, MoabCluster, SLURMCluster,
+                                     SGECluster, LSFCluster])
+@pytest.mark.parametrize(
```

Comment:
General remark, for the error cases I would remove this parametrize and just test with something like this:
* an output without match: e.g. `'there is no number here'`
* a regexp that match but that does not have a `job_id` named group, e.g. `cluster.job_id_regexp = r'\d+'` and an output `'Job <12345> is submitted to default queue <normal>.`